graylog2-server/src/test/java/org/graylog2/security/PasswordAlgorithmFactoryTest.java

```diff
@@ -0,0 +1,93 @@
+/**
+ * This file is part of Graylog.
+ *
+ * Graylog is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Graylog is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Graylog.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.graylog2.security;
+
+import com.google.common.collect.ImmutableMap;
+import org.graylog2.plugin.security.PasswordAlgorithm;
+import org.graylog2.security.PasswordAlgorithmFactory;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import java.util.Collections;
+import java.util.Map;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class PasswordAlgorithmFactoryTest {
+
+    @Mock
+    private PasswordAlgorithm passwordAlgorithm1;
+    @Mock
+    private PasswordAlgorithm passwordAlgorithm2;
+
+    private Map<String, PasswordAlgorithm> passwordAlgorithms;
+
+    @Before
+    public void setUp() throws Exception {
+        this.passwordAlgorithms = ImmutableMap.<String, PasswordAlgorithm>builder()
```

Comment:
Since the map is immutable, this could be done once (and maybe even change `passwordAlgorithms` to a constant, i. e. `static final`).